components/src/core/Drawer/DrawerRailItem.tsx

```diff
@@ -0,0 +1,259 @@
+            left: 0,
+            width: '100%',
+        },
+        icon: {
+            color: (props): string =>
+                props.itemID !== props.activeItem
+                    ? props.itemIconColor || theme.palette.text.primary
+                    : props.activeItemIconColor ||
+                      (theme.palette.type === 'light' ? theme.palette.primary.main : lightenedPrimary),
+            backgroundColor: 'transparent',
+            height: 'auto',
+            width: 'auto',
+        },
+        statusStripe: {
+            position: 'absolute',
+            top: 0,
+            bottom: 0,
+            left: 0,
+            width: 6,
+            zIndex: 100,
+            backgroundColor: (props): string => props.statusColor,
+        },
+        title: {
+            lineHeight: `16px`,
+            wordBreak: 'break-word',
+            hyphens: 'auto',
+            zIndex: 200,
+            color: (props): string =>
+                props.itemID !== props.activeItem
+                    ? props.itemFontColor || theme.palette.text.primary
+                    : props.activeItemFontColor ||
+                      (theme.palette.type === 'light' ? theme.palette.primary.main : lightenedPrimary),
+        },
+        titleActive: {
+            fontWeight: 600,
+        },
+        ripple: {
+            backgroundColor: theme.palette.primary.main,
+        },
+    });
+});
+
+const DrawerRailItemRender: React.ForwardRefRenderFunction<unknown, DrawerRailItemProps> = (
+    props: DrawerRailItemProps,
+    ref: any
+) => {
+    const {
+        activeItem,
+        classes = {},
+        condensed = false,
+        hidden,
+        icon,
+        itemID,
+        onClick,
+        onItemSelect,
+        ripple = true,
+        title,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+        activeItemBackgroundColor,
+        activeItemFontColor,
+        activeItemIconColor,
+        divider,
+        itemFontColor,
+        itemIconColor,
+        statusColor,
+        /* eslint-disable @typescript-eslint/no-unused-vars */
+
+        ...buttonBaseProps
+    } = props;
+
+    const defaultClasses = useStyles(props);
+
+    const active = activeItem === itemID;
+    const hasAction = Boolean(onItemSelect || onClick);
+
+    // Customize the color of the Touch Ripple
+    const RippleProps =
+        ripple && hasAction
+            ? {
+                  TouchRippleProps: {
+                      classes: {
+                          child: defaultClasses.ripple,
+                      },
+                  },
+              }
+            : {};
+
+    const combine = useCallback(
+        (className: keyof DrawerRailItemClasses): string => clsx(defaultClasses[className], classes[className]),
+        [defaultClasses, classes]
+    );
+
+    const getIcon = useCallback((): JSX.Element | undefined => {
+        if (icon) {
+            return <Avatar className={combine('icon')}>{icon}</Avatar>;
+        }
```

Comment:
Somehow when I try to wrap icons in a Badge, the badge doesn't show up normally. I suppose we are not going to use an Avatared icon for navigation rail anyway, so maybe we should abandon this avatar logic? 

![image](https://user-images.githubusercontent.com/8997218/104697444-c357bb80-56dd-11eb-8b76-703927e25c0a.png)

This corresponds to the code 

``` tsx
                        {
                            title: titleList[0],
                            itemID: titleList[0],
                            icon: <Badge color={'primary'} badgeContent={'2'}><ListIcon className={clsx({ [classes.iconFlip]: rtl })} /></Badge>,
                            onClick: (): void => {
                                navigate(0);
                                if (xsDown) dispatch({ type: CLOSE_DRAWER });
                            },
                            statusColor: 'red',
                        },
```

in showcase

Reply:
The Avatar is there to maintain a consistent sizing. I can turn on overflow so you can see a badge that overextends the size.

Reply:
That works too. The badges should work out of the box